### PR TITLE
Add support for storing execution stats per nbformat

### LINKED ISSUE #2660: Support execution times in notebook per nbformat v4.4
As [stated in the spark roadmap](https://github.com/jupyter/roadmap/blob/master/spark.md#timing-data), we want to store how long it took to run a cell. The transient information for how long execution took is available in the message spec.

[JupyterLab has started adopting it with the same metadata the `execute_time` extension was using](https://github.com/jupyterlab/jupyterlab/pull/4161#issuecomment-371929527).

### PR DESCRIPTION
**Pull request by captainsafia**
Closes https://github.com/nteract/nteract/issues/2660.

### CODE REVIEW

File: packages/epics/src/execute.ts [hunk 1]
```diff
@@ -85,6 +85,54 @@ export function executeCellStream(
...
+     * Set the ISO datetime when the execute_input message
+     * was broadcast from the kernel, per nbformat.
+     */
+    cellMessages.pipe(
+      ofMessageType("execute_input"),
+      map(() =>
+        actions.setInCell({
+          id,
+          contentRef,
+          path: ["metadata", "execution", "iopub", "execute_input"],
```
Comment (MSeal) [inline packages/epics/src/execute.ts]: FYi in the spec I believe the key name at the end should be `iopub.execute_input` and not another map with `iopub` key:

https://github.com/jupyter/nbformat/blob/master/nbformat/v4/nbformat.v4.4.schema.json#L212-L227

Comment (MSeal) [reply packages/epics/src/execute.ts]: Same for the other settesr

**Review by MSeal — DISMISSED** [review]
Unless I am mistaken, while the way described here seemed more intuitive from a object store pov it differs from the spec slightly.

**Comment by captainsafia** [comment]
> Unless I am mistaken, while the way described here seemed more intuitive from a object store pov it differs from the spec slightly.

Ah! This makes sense. I was referring to the Sphinx docs when I was making this PR and my JS-leaning brain interpreted the periods as indicators for a nested object. I can update this to be string keys though.

File: packages/epics/src/execute.ts (hunk 1 above)
Comment (rgbkrk) [reply packages/epics/src/execute.ts]: What a strange way to name a key in a json spec. 🙄

**Review by rgbkrk — APPROVED** [review]
Since this matches the spec with the new commit, this looks good.

File: packages/epics/src/execute.ts (hunk 1 above)
Comment (MSeal) [reply packages/epics/src/execute.ts]: Yeah, I didn't catch that when it was PR'd in nbformat -- I can see it making sense from a namespace -> key, value pair POV where you want a fixed depth of settings. But I agree it does make runtime path assignments awkward.

**Pull request merged**